### DISCUSSION TITLE
Isometric occlusion

### DISCUSSION THREAD

**Discussion by jumpjack** [melonjs/melonJS#1161] [Q&A]
I see that in this example:

https://github.com/jkalkhof/urban-survivor-rpg

the occlusion is made by placing the base of an object in a layer and the remaining part of the object in another layer, and placing the player layer between them.

![image](https://user-images.githubusercontent.com/1620953/203523351-608fe5e9-e5cb-478f-a33b-1180d8c6b8d6.png)

![image](https://user-images.githubusercontent.com/1620953/203523400-b14a2fee-ebcf-4196-8a12-952056cc41b7.png)

Is this the only possible method in MelonJS?
I have an existing tile map which I  cannot modify, coming from a game where occlusion is obtained by draw order, and big objects are made of tall tiles; for example: the base tile dimension is 24x12, but if I have a wall, it's 24x48 (i.e. 3 tiles tall), so the player can move around the 24x12 base of the big tile, but when it's above the base tile, it's actually hidden behind the 2 remaining vertical subtiles of the  wall.

![image](https://user-images.githubusercontent.com/1620953/203523112-6533e3a2-efd4-4c72-b495-bcadd0eb2f0d.png)

**Comment by obiot** [comment]
melonJS will actually follow the `draworder` and `renderorder` properties as defined in Tiled, while the latter is only for orthogonal maps (if I recall correctly) the former is the one you are looking after.

you can also dynamically change the sort order (which is the draw order) of any container ( like game.world) by using this (it's `z` by default) :
https://melonjs.github.io/melonJS/docs/melonjs/Container.html#sortOn
